keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlAgentLockRepository.kt

```diff
@@ -0,0 +1,72 @@
+package com.netflix.spinnaker.keel.sql
+
+import com.netflix.spinnaker.keel.persistence.AgentLockRepository
+import com.netflix.spinnaker.keel.persistence.metamodel.Tables.AGENT_LOCK
+import com.netflix.spinnaker.keel.scheduled.ScheduledAgent
+import com.netflix.spinnaker.kork.sql.config.RetryProperties
+import io.github.resilience4j.retry.Retry
+import io.github.resilience4j.retry.RetryConfig
+import io.vavr.control.Try
+import java.time.Clock
+import java.time.Duration
+import java.util.concurrent.TimeUnit
+import org.jooq.DSLContext
+import org.jooq.exception.SQLDialectNotSupportedException
+import org.springframework.scheduling.annotation.Scheduled
+
+class SqlAgentLockRepository(
+  private val jooq: DSLContext,
+  private val clock: Clock,
+  private val retryProperties: RetryProperties,
+  private val agents: List<ScheduledAgent>
+) : AgentLockRepository {
+
+  @Scheduled(fixedDelayString = "\${keel.scheduled.agent.frequency:PT1m}")
+  fun invoke() {
+    agents.forEach {
+      val lockAcquired = tryAcquireLock(it.javaClass.simpleName, TimeUnit.MINUTES.toSeconds(5))
```

Comment:
Since lock time has the effect of setting execution frequency, I think it should be a property on `ScheduledAgent` that can be overridden by implementing classes.